### ISSUE TITLE
Bug on bulkCreate() with sync({ alter: true })

### ISSUE THREAD

**Issue by mahanhaz** [sequelize/sequelize#7974]
Basicly I'm trying to insert bulk of data into my table.
```js
sequelize.sync({alter: true})
```

following code represent my action:
```js
    const expiry = now.setDate(now.getDate() + config.invoice.defaultValues.expiryDate)

    for (let recipient of recipients) {
      invoices.push({
        accountName: 'test',
        amount: 50,
        expiryDate: expiry
      })
    }

    const records = await sequelize.invoice.bulkCreate(invoices)
```
following code represents my model:
```js
'use strict'

import config from '../config'

export default function (sequelize, DataTypes) {
  const Invoice = sequelize.define('invoice', {
    accountName: {
      type: DataTypes.STRING,
      field: 'account_name',
    },
    amount: {
      type: DataTypes.FLOAT,
      field: 'amount',
    },
    expiryDate: {
      type: DataTypes.DATE,
      field: 'expiry_date',
    },
  }, {
    tableName: 'invoice',
    createdAt: 'created',
    updatedAt: 'updated'
  })

  return Invoice
}
```
as you can see i have alias name for my column names which i used `field` in order to have deferent column name in my mysql database for following columns: `accountName`, `expiryDate`.

## What do i expect to happen?
these codes work perfectly when `alter` is `false` the same `true`

## What is actually happening?
when i turn alter to true, in addition to execution of queries, it alter the table with index of my object instead of defined alias

this is the log of the sequelize on `alter: false`
```
Executing (default): CREATE TABLE IF NOT EXISTS `invoice` (`id` INTEGER NOT NULL auto_increment , `account_name` VARCHAR(255), `amount` FLOAT, `expiry_date` DATETIME, `created` DATETIME NOT NULL, `updated` DATETIME NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB;
Executing (default): INSERT INTO `invoice` (`id`,`account_name`,`amount`,`expiry_date`,`created`,`updated`) VALUES (NULL,'1234','1234','2017-08-17 11:00:34','2017-07-18 11:00:34','2017-07-18 11:00:34');
```

and this is the log on `alter: true`:
```
Executing (default): CREATE TABLE IF NOT EXISTS `invoice` (`id` INTEGER NOT NULL auto_increment , `account_name` VARCHAR(255), `amount` FLOAT, `expiry_date` DATETIME, `created` DATETIME NOT NULL, `updated` DATETIME NOT NULL, PRIMARY KEY (`id`)) ENGINE=InnoDB;
Executing (default): INSERT INTO `invoice` (`id`,`account_name`,`amount`,`expiry_date`,`created`,`updated`) VALUES (NULL,'1234','1234','2017-08-17 11:00:34','2017-07-18 11:00:34','2017-07-18 11:00:34');
Executing (default): ALTER TABLE `invoice` ADD `accountName` VARCHAR(255);
Executing (default): ALTER TABLE `invoice` ADD `expiryDate` DATETIME;
Executing (default): ALTER TABLE `invoice` DROP `account_number`;
Executing (default): SELECT CONSTRAINT_NAME as constraint_name FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE WHERE (REFERENCED_TABLE_NAME = 'invoice' AND REFERENCED_COLUMN_NAME = 'account_name') OR (TABLE_NAME = 'invoice' AND COLUMN_NAME = 'account_name')
Executing (default): ALTER TABLE `invoice` DROP `account_name`;
Executing (default): ALTER TABLE `invoice` CHANGE `amount` `amount` FLOAT;
Executing (default): SELECT CONSTRAINT_NAME as constraint_name FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE WHERE (REFERENCED_TABLE_NAME = 'invoice' AND REFERENCED_COLUMN_NAME = 'expiry_date') OR (TABLE_NAME = 'invoice' AND COLUMN_NAME = 'expiry_date')
Executing (default): ALTER TABLE `invoice` DROP `expiry_date`;
Executing (default): ALTER TABLE `invoice` CHANGE `created` `created` DATETIME NOT NULL;
Executing (default): ALTER TABLE `invoice` CHANGE `updated` `updated` DATETIME NOT NULL;
```

basically it changes the schema of table after a simple `bulkCreate` to a wrong structure and totally ignores the column field in the model

```
$ node -v
v8.1.2

$ uname -a
Linux 32cd23d445e0 4.9.36-moby #1 SMP Wed Jul 12 15:29:07 UTC 2017 x86_64 GNU/Linux
```

__Dialect:__ mysql
__Database version:__ 5.6
__Sequelize version:__ 4.3.1

**Comment by tomgao365** [comment]
We have the same problem

**Comment by tomgao365** [comment]
Hi, @mahanhaz 
I found bugs here：
https://github.com/sequelize/sequelize/blob/master/lib/model.js#L1109
